src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,59 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.constructor.name !== 'Herbivore') {
+      Animal.alive.push(this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+
+    Animal.alive = Animal.alive.map((objAnimal) => {
+      if (objAnimal.name === this.name) {
+        return { ...objAnimal, hidden: !objAnimal.hidden };
+      } else {
+        return objAnimal;
+      }
+    });
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal.constructor.name !== 'Carnivore' && animal.hidden !== true) {
+      Animal.alive = Animal.alive
```

Comment:
Removing the animal from `Animal.alive` is a herbivore's behavior thus, it would make sense to allow the `Herbivore` class handle it and not move into the `Carnivore` class, could you come up with a way to achieve this?

Tip: remember getters and setters

**This is a personal suggestion/challenge, doing it this way is not required**